From: Mohamad Al-Jaf <mohamadaljaf@gmail.com>
Subject: [PATCH v7 3/5] wdscore: Add stubs for ConstructPartialMsgV[AW].
Message-Id: <20220211010215.183806-3-mohamadaljaf@gmail.com>
Date: Thu, 10 Feb 2022 20:02:13 -0500
In-Reply-To: <20220211010215.183806-1-mohamadaljaf@gmail.com>
References: <20220211010215.183806-1-mohamadaljaf@gmail.com>

Enables the Windows MediaCreationTool21H2 to run.

Signed-off-by: Mohamad Al-Jaf <mohamadaljaf@gmail.com>
---
v7: Fix incorrect enum values.
---
 dlls/wdscore/main.c             | 18 ++++++++++++++++++
 dlls/wdscore/wdscore.spec       |  4 ++--
 dlls/wdscore/wdscore_internal.h | 11 +++++++++++
 3 files changed, 31 insertions(+), 2 deletions(-)

diff --git a/dlls/wdscore/main.c b/dlls/wdscore/main.c
index 11111111111..11111111111 100644
--- a/dlls/wdscore/main.c
+++ b/dlls/wdscore/main.c
@@ -54,3 +54,21 @@ LPVOID CurrentIP(void)
     return NULL;
 }
 #endif
+
+/***********************************************************************
+ *           ConstructPartialMsgVA (wdscore.@)
+ */
+LPVOID WINAPI ConstructPartialMsgVA( WdsLogLevel level, LPCSTR msg, va_list args )
+{
+    FIXME( "%u %s - stub\n", level, debugstr_a(msg) );
+    return NULL;
+}
+
+/***********************************************************************
+ *           ConstructPartialMsgVW (wdscore.@)
+ */
+LPVOID WINAPI ConstructPartialMsgVW( WdsLogLevel level, LPCWSTR msg, va_list args )
+{
+    FIXME( "%u %s - stub\n", level, debugstr_w(msg) );
+    return NULL;
+}
diff --git a/dlls/wdscore/wdscore.spec b/dlls/wdscore/wdscore.spec
index 11111111111..11111111111 100644
--- a/dlls/wdscore/wdscore.spec
+++ b/dlls/wdscore/wdscore.spec
@@ -69,8 +69,8 @@
 #@ extern g_bEnableDiagnosticMode
 @ stub ConstructPartialMsgIfA
 @ stub ConstructPartialMsgIfW
-@ stub ConstructPartialMsgVA
-@ stub ConstructPartialMsgVW
+@ stdcall ConstructPartialMsgVA(long str ptr)
+@ stdcall ConstructPartialMsgVW(long wstr ptr)
 @ cdecl CurrentIP()
 @ stub EndMajorTask
 @ stub EndMinorTask
diff --git a/dlls/wdscore/wdscore_internal.h b/dlls/wdscore/wdscore_internal.h
index 11111111111..11111111111 100644
--- a/dlls/wdscore/wdscore_internal.h
+++ b/dlls/wdscore/wdscore_internal.h
@@ -19,6 +19,17 @@
 #ifndef __WDSCORE_INTERNAL_H
 #define __WDSCORE_INTERNAL_H
 
+typedef enum _WdsLogLevel {
+    WdsLogLevelAssert     = 0x00000000,
+    WdsLogLevelFatalError = 0x01000000,
+    WdsLogLevelError      = 0x02000000,
+    WdsLogLevelWarning    = 0x03000000,
+    WdsLogLevelInfo       = 0x04000000,
+    WdsLogLevelStatus     = 0x05000000,
+    WdsLogLevelVerbose    = 0x68000000,
+    WdsLogLevelTrace      = 0x70000000
+} WdsLogLevel;
+
 LPVOID CurrentIP(void);
 
 #endif /* __WDSCORE_INTERNAL_H */

-- 
2.35.1

